yellowbrick/text/postag.py

```diff
@@ -114,251 +115,381 @@ def __init__(
+                list(self.pos_tag_counts_[self.labels_[i]].values()),
+                bottom=last,
+                color=colors[i] 
+                )
+                last = [sum(x) for x in zip(list(last), 
+                         list(self.pos_tag_counts_[self.labels_[i]].values()))]
+        else:
+            # Plot barplot(not stacked)
+            colors = resolve_colors(
+                n_colors=len(self.pos_tag_counts_), 
+                colormap=self.colormap, 
+                colors=self.colors
+                )
+            if self.frequency:
+                # Sort tags with respect to frequency in corpus
+                self.sorted_tags = sorted(
+                    self.pos_tag_counts_, key=self.pos_tag_counts_.get, 
+                    reverse=True
+                    )
+                sorted_counts = [self.pos_tag_counts_[tag] for tag in self.sorted_tags]
+    
+                self.ax.bar(range(len(self.sorted_tags)), sorted_counts, 
+                            color=colors)
+                return self.ax
+            
+            self.ax.bar(
+                range(len(self.pos_tag_counts_)),
+                list(self.pos_tag_counts_.values()),
+                color=colors,
+                )
         return self.ax
+        
 
     def finalize(self, **kwargs):
         """
         Finalize the plot with ticks, labels, and title
-
         Parameters
         ----------
         kwargs: dict
             generic keyword arguments.
         """
         # NOTE: not deduping here, so this is total, not unique
-        self.set_title(
-            "PosTag plot for {}-token corpus".format(
-                (sum(self.pos_tag_counts_.values()))
-            )
-        )
-
-        self.ax.set_xticks(range(len(self.pos_tag_counts_)))
-        self.ax.set_xticklabels(list(self.pos_tag_counts_.keys()), rotation=90)
-
-        # Set the axis labels
-        if self.frequency:
-            self.ax.set_xlabel(
-                "{} part-of-speech tags, sorted by frequency".format(self.tagset_names[self.tagset])
+        
+        self.ax.set_ylabel("Count")
+        if self.stack:
+            self.ax.set_xticks(range(len(self.pos_tag_counts_[self.labels_[0]])))
+            if self.frequency:
+                self.ax.set_xticklabels(self.sorted_tags, rotation=90)
+            else:
+                self.ax.set_xticklabels(list(self.pos_tag_counts_[self.label_[0]].keys()),
+                                    rotation=90)
+            colors = resolve_colors(
+                    n_colors=len(self.labels_), 
+                    colormap=self.colormap,
+                    colors=self.colors
+                    )
+            manual_legend(self, self.labels_, colors, loc='best')
+            self.set_title(
+                "PosTag plot for {}-token corpus".format(
+                    (sum([sum(i.values()) for i in self.pos_tag_counts_.values()]))
+                )
```

Comment:
As in `draw()` the control flow logic is at an appropriate level, but there is still a lot of duplicate code. Would you mind either refactoring to minimize the amount of duplicated code or to specialize the functionality for each case and comment which case is which?